wiki/mapping/map-format.md

```diff
@@ -313,16 +313,456 @@ The exact specifics of what goes in `_customData` is entirely dependent on commu
+  "obstacles": [],
+  "sliders": [],
+  "burstSliders": [],
+  "waypoints": [],
+  "basicBeatmapEvents": [],
+  "colorBoostBeatmapEvents": [],
+  "lightColorEventBoxGroups": [],
+  "lightRotationEventBoxGroups": [],
+  "basicEventTypesWithKeywords": [],
+  "useNormalEventsAsCompatibleEvents": false
+}
+```
+
+#### version
+This field describes the version of the map format we are using.
+
+#### bpmEvents
+This is an array of [BPM Event](#bpm-events) objects for the map.
+
+#### rotationEvents
+This is an array of [Rotation Event](#rotation-events) objects for the map.
+
+#### colorNotes
+This is an array of [Color Note](#color-notes) objects for the map.
+
+#### bombNotes
+This is an array of [Bomb Note](#bomb-notes) objects for the map.
+
+#### obstacles
+This is an array of [Obstacle](#obstacles-2) objects for the map.
+
+#### sliders
+This is an array of [Slider](#sliders-2) objects for the map.
+
+#### burstSliders
+This is an array of [Burst Slider](#burst-sliders) objects for the map.
+
+#### waypoints
+::: tip NOTE
+This is a stub section.
+:::
+This is used to control BTS TinyTAN figures. Some information can be found in this
+[document](https://docs.google.com/spreadsheets/d/1spW7LS-RvenLQBVXJl9w_iOwqr9r_ozxYo3JUlXq9Lc).
+
+#### basicBeatMapEvents
+This is an array of [Basic Event](#basic-beatmap-events) objects for the map.
+
+#### colorBoostBeatmapEvents
+This is an array of [Boost Event](#color-boost-beatmap-events) objects for the map.
+
+#### lightColorEventBoxGroups
+This is an array of Light Color Group Event objects for the map.
+::: tip NOTE
+This is a stub section. Documentation is a work in progress.
+:::
+
+#### lightRotationEventBoxGroups
+This is an array of Light Rotation Group Event objects for the map.
+::: tip NOTE
+This is a stub section. Documentation is a work in progress.
+:::
+
+#### basicEventTypesWithKeywords
+::: tip NOTE
+This is a stub section. Documentation is a work in progress.
+:::
+  
+#### useNormalEventsAsCompatibleEvents
+This is a boolean.
+::: tip NOTE
+This is a stub section. Documentation is a work in progress.
+:::
+
+### Bpm Events
+
+```json
+{
+  "b": 10.0,
+  "m": 128.0,
+}
+```
+
+:::danger
+As of Beat Saber `1.20.0`, Official BPM Changes are still not complete. An official mapper has advised against using this
+event in its current state.
+:::
+
+#### b
+The time, in beats, where this object reaches the player.
+
+#### m
+A float representing the new bpm. Any `b` in objects after this point will be adjusted to the new bpm.
+
+### Rotation Events
+
+```json
+{
+  "b": 10.0, // Beat
+  "e": 0,    // Event type
+  "r": 15.0, // Rotation
+}
+```
+
+#### b
+The time, in beats, where this object reaches the player.
+
+#### e
+|`e`|Result|
+|:---:|-----|
+|`0`|Early rotation. Rotates future objects, while also rotating objects at the same time.|
+|`1`|Late rotation. Rotates future objects, but ignores rotating objects at the same time.|
+
+#### r
+A float which represents clockwise rotation.
```

Comment:
Small nitpick, but you might want to specify if that's viewed from above or below (I'm assuming above).